### PR TITLE
First Try

### PR DESCRIPTION
**Pull request by marsaud**
ES2016 Syntax, not executable as is, only for read and code review

### CODE REVIEW

File: lib/api/createReadStream.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      size -= packetSize
+      if (this.encoding) {
+        content = content.toString(this.encoding)
+      }
+      if (!this.push(content)) {
+        return
+      }
+    }
+    if (this.offset.lt(this.fileLength)) {
+      await SMB2RequestAsync('close', this.file, this.connection)
```
Comment (julien-f) [inline lib/api/createReadStream.js]: I think you do not need to wait for the completion of this operation.

And you should ignore any errors (`.catch(noop)`).

File: lib/api/createReadStream.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      this.push(null)
+    }
+  }
+}
+
+export default async function (path, options = {}) {
+  const file = await SMB2RequestAsync('open', {path}, this)
+  options.objectMode = false
```
Comment (julien-f) [inline lib/api/createReadStream.js]: Why is it needed?

**Comment by julien-f** [comment]
Nice commits messages :p

**Comment by marsaud** [comment]
Yeah, went too fast :) All is rebased and forced now. I know, this is butchery, but the room is clear enough to avoid accidental amputation ^^

**Pull request merged**